src/neumann.jl

```diff
@@ -0,0 +1,58 @@
+# If face_value is ScalarValued, f should return scalar 
+# If face_value is VectorValued, f should return a vector
+struct Neumann{FV}      # cells and faces are a deconstructed
+    cells::Vector{Int}  # FaceIndex, to allow loop over cells 
+    faces::Vector{Int}  # with CellIterator 
+    face_values::FV     
+    field_name::Symbol
+    fun # f(x::Vec, t::Number, normal::Vec)->Union{<:Number, <:Vec}
```

Comment:
`fun` will be abstractly typed, as each function has its own type. Perhaps it should be a type parameter?

Reply:
Yes, I have planned to do that, but right now I let it mirror the Dirichlet (except the restriction of ::Function).